### PR TITLE
api: fill data in column lineage input nodes.

### PR DESCRIPTION
**Pull request by JDarDagran**
### Problem

As stated in #2725 data is null for output nodes in column lineage endpoint.

### Solution

Fill data field.

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

**Review by pawel-big-lebowski — APPROVED** [review]

**Review by pawel-big-lebowski — APPROVED** [review]

File: api/src/main/java/marquez/db/models/ColumnLineageNodeData.java [hunk 1]
```diff
@@ -42,6 +42,17 @@ public ColumnLineageNodeData(
...
+  public ColumnLineageNodeData(InputFieldNodeData data) {
+    this.namespace = data.namespace;
+    this.dataset = data.dataset;
+    this.datasetVersion = data.datasetVersion;
+    this.field = data.field;
+    this.fieldType = "";
+    this.transformationDescription = data.transformationDescription;
+    this.transformationType = data.transformationType;
+    this.inputFields = ImmutableList.of();
+  }
```
Comment (phixMe) [inline api/src/main/java/marquez/db/models/ColumnLineageNodeData.java:54]: I'm a little confused by some of these fields like `fieldType` and `inputFields`. Would these ever be set here?

Comment (JDarDagran) [reply api/src/main/java/marquez/db/models/ColumnLineageNodeData.java:54]: The truth is that if I set `InputFieldNodeData` to implement `NodeData` interface jackson couldn't deduce what was type of `NodeData` was in the output node (the one that previously had `data` field null). For that reason I basically made a workaround to simply create `ColumnLineageNodeData` from `InputFieldNodeData`.

I'd need more guidance how to fix jackson deduction :|

**Comment by tatiana** [comment]
@wslulciuc @phixMe - while talking to @JDarDagran, he mentioned that there were concerns this PR may not solve the original issue and that you'd be logging an issue with further details. Could you confirm if that is the case?

**Comment by wslulciuc** [comment]
@tatiana: yeah, although this is a "fix", I think the issue is more fundemtail to our colum lineage query. That is, @phix and I looked into this a few days ago and walked through every OL call (@JDarDagran provided us) and it was enlightening. I’d like to share what we found, just haven’t had a window of time to write up our findings but the tl;dr is: the **output_dataset_field_uuid** in our [column lineage query](https://github.com/MarquezProject/marquez/blob/main/api/src/main/java/marquez/db/ColumnLineageDao.java#L169) doesn’t return values for the **ColumnLineageNode**s on the edges of the graph (ie. graph nodes with no input columns, see below). I’d like to further confirm this, but given that our seed data works just fine, there’s something we are not accounting for.

IMG_6440

**Comment by pawel-big-lebowski** [comment]
> @tatiana: yeah, although this is a "fix", I think the issue is more fundemtail to our colum lineage query. That is, @phix and I looked into this a few days ago and walked through every OL call (@JDarDagran provided us) and it was enlightening. I’d like to share what we found, just haven’t had a window of time to write up our findings but the tl;dr is: the **output_dataset_field_uuid** in our [column lineage query](https://github.com/MarquezProject/marquez/blob/main/api/src/main/java/marquez/db/ColumnLineageDao.java#L169) doesn’t return values for the **ColumnLineageNode**s on the edges of the graph (ie. graph nodes with no input columns, see below). I’d like to further confirm this, but given that our seed data works just fine, there’s something we are not accounting for.
> 

Hi @wslulciuc, I spent some time investigating nulls and I think @JDarDagran's PR solves the problem we were able to reproduce. The PR modifies a test and then applies changes need to fix it. That's why I think it is worth merging from that perspective. 

What do you say? Once you provide more details on other issue, we can resume discussion while not blocking this.

**Comment by wslulciuc** [comment]
@pawel-big-lebowski yeah, I'll merge 😉. I'm fine with that as we don't want to further delay the merge of https://github.com/MarquezProject/marquez/pull/2725. Mind opening an issue to dive deep into the column-level lineage query and move our discussion there?

**Review by wslulciuc — APPROVED** [review]
Thanks for the fix, @JDarDagran! We'll want to open an issue to dive deeper into the column-level lineage query and move our discussion there. But, giving this a +1 to not block https://github.com/MarquezProject/marquez/pull/2742

**Pull request merged**